omero/sysadmins/client-server-ssl.rst

```diff
@@ -0,0 +1,90 @@
+Client Server SSL verification
+==============================
+
+If you configure OMERO.web behind Nginx with a recognized SSL certificate your users can be sure that they are connecting to their intended server.
+
+OMERO.server and clients do not automatically support host verification, so a
+`man-in-the-middle attack <https://www.cloudflare.com/learning/security/threats/man-in-the-middle-attack/>`_
+is possible.
+This may result in users inadvertently transmitting their login credentials to an attacker.
+
+This can be remedied by configuring OMERO.server with a certificate (the same certificate used for OMERO.web Nginx may work), and ensuring all OMERO clients are configured to verify the server certificate before connecting.
+
+
+Server certificate
+------------------
+
+The easiest solution is to re-use the SSL certificates used to protect OMERO.web.
+First convert the public certificate ``server.pem`` and private key ``server.key`` to the PKCS12 format where ``secret`` is the password used to protect the combined output file ``server.p12``::
+
+    openssl pkcs12 -export -out server.p12 -in server.pem -inkey server.key -passout pass:secret
+
+Copy ``server.p12`` to the OMERO.server host, for instance to ``/etc/ssl/omero/``.
+
+External access to OMERO.server is managed by the Glacier2 component which can be configured as follows::
+
+    # Enable authenticating ciphers.
+    omero config set omero.glacier2.IceSSL.Ciphers "ADH:HIGH:!LOW:!MD5:!EXP:!3DES:@STRENGTH"
+    # Look for certificates in this directory, you can omit and use the full path to files instead
+    omero config set omero.glacier2.IceSSL.DefaultDir /etc/ssl/omero/
+    omero config set omero.glacier2.IceSSL.CertFile server.p12
+    omero config set omero.glacier2.IceSSL.Password secret
+
+For even stronger security require TLS 1.2, disable anonymous ciphers and only allow ``HIGH``::
+
+    omero config set omero.glacier2.IceSSL.Protocols tls1_2
+    omero config set omero.glacier2.IceSSL.ProtocolVersionMin tls1_2
+    omero config set omero.glacier2.IceSSL.ProtocolVersionMax tls1_2
```

Comment:
Is setting a maximum beneficial?

Reply:
It's set to `tls1_1`: https://github.com/openmicroscopy/openmicroscopy/blob/v5.5.0-m5/etc/templates/grid/templates.xml#L485 and I'm not sure what happens if `Protocols` or `ProtocolVersionMin` if greater than `ProtocolVersionMax`

Reply:
Ahh, it must be set to *something*? Makes perfect sense, ta.

Reply:
More that it's set by default, and I can't think of an easy way to unset it.